xrpl/wallet/main.py

```diff
@@ -16,69 +17,131 @@ class Wallet:
+
+    address = classic_address
+    """
+    The address that publicly identifies this wallet, as a base58 string.
+    """
+
     def __init__(
         self: Wallet,
-        seed: str,
-        sequence: int,
+        public_key: str,
+        private_key: str,
         *,
-        algorithm: Optional[CryptoAlgorithm] = None,
+        master_address: Optional[str] = None,
+        seed: Optional[str] = None,
     ) -> None:
         """
         Generate a new Wallet.
 
         Args:
-            seed: The seed from which the public and private keys are derived.
-            sequence: The next sequence number for the account.
-            algorithm: The algorithm used to encode the keys. Inferred from the seed if
-                not included.
-        """
-        self.seed = seed
-        """
-        The core value that is used to derive all other information about
-        this wallet. MUST be kept secret!
+            public_key: The public key for the account.
+            private_key: The private key used for signing transactions for the account.
+            master_address: Include if a Wallet uses a Regular Key Pair. It must be
+                the master address of the account. The default is `None`.
+            seed: The seed used to derive the account keys. The default is `None`.
         """
-
-        pk, sk = derive_keypair(self.seed, algorithm=algorithm)
-        self.public_key = pk
+        self.public_key = public_key
         """
         The public key that is used to identify this wallet's signatures, as
         a hexadecimal string.
         """
 
-        self.private_key = sk
+        self.private_key = private_key
         """
         The private key that is used to create signatures, as a hexadecimal
         string. MUST be kept secret!
         """
 
-        self.classic_address = derive_classic_address(self.public_key)
-        """The address that publicly identifies this wallet, as a base58 string."""
+        self._classic_address = (
+            ensure_classic_address(master_address)
+            if master_address is not None
+            else derive_classic_address(self.public_key)
+        )
+        """Internal variable for classic_address. Use classic_address instead."""
 
-        self.sequence = sequence
+        self.seed = seed
         """
-        The next available sequence number to use for transactions from this
-        wallet.
-        Must be updated by the user. Increments on the ledger with every successful
-        transaction submission, and stays the same with every failed transaction
-        submission.
+        The core value that is used to derive all other information about
+        this wallet. MUST be kept secret!
         """
 
     @classmethod
-    def create(
-        cls: Type[Wallet], crypto_algorithm: CryptoAlgorithm = CryptoAlgorithm.ED25519
+    def generate(
+        cls: Type[Wallet], algorithm: CryptoAlgorithm = CryptoAlgorithm.ED25519
```

Comment:
I think this doesn't need to be renamed - `create` seems like just as good a name as `generate` in my opinion.

Reply:
Ok, was weighing between keeping similar to xrpl.js vs keeping as similar to old xrpl-py

Reply:
My thought process is that since the goal is to make xrpl-py better we want to make a good design that is easy to transition from wherever possible. (So since this is a neutral change in terms of usability, sticking with xrpl-py convention is preferred)